### ISSUE TITLE
Timer counting down went very fast after the clock is blocked by a modal dialog

### ISSUE THREAD

**Issue by jenkins-teamone** [pughpugh/react-countdown-clock#39]
In my code, I pop up a modal dialog in page when a button is clicked that blocks the view of clock. After the modal dialog is dismissed, the counting down time went fast and time even jump back. Do I need to pause the clock before display things on top of it?

**Comment by pughpugh** [comment]
Hi. I've not seen such a case that would warrant stopping the clock.  If possible, you'd have to provide an example to replicate the issue, as it's hard to say what's going on from just a brief description.

**Comment by PierreAndreis** [comment]
I had this problem.
My countdown was going too fast after a callback from a second countdown on the same page was firing the fast paced countdown to start (which was paused from true to false).

I rewrote some of the code on my end, and this is what my shouldComponentUpdate looks like:
`  componentDidUpdate(props) {
    let changes = false;
    if (props.seconds !== this.props.seconds) {
      changes = true;
      this._seconds = props.seconds;
    }

    if (props.paused !== this.props.paused) {
      changes = true;
    }
    if (changes) {
      this._setupTimer();
    }
  };`

and this is my _setupTimer:
`
  _setupTimer() {

    if (this._canvas) {
      this._clearBackground();
    }

    this._setScale();
    this._setupCanvases();
    this._drawBackground();
    this._drawTimer();

    if (!this.props.paused) {
      return this._startTimer();
    }
    this._pauseTimer();
  };
`

PS: I migrated to Class ES6 and ES6.

**Comment by cyang78** [comment]
I had this problem, too. I'm using antd https://ant.design/components/modal/ to pop up the modal dialog. 
My snippet is like this:

<div>
             <RequestInfo {...requestInfoProps} />
             <Modal
        visible={modalDialogVisible}
        title="Accept Request"
        onOk={acceptRequestOnOK}
        onCancel={acceptRequestOnCancel}
        footer={[
          <Button
            key="back" size="large"
            onClick={acceptRequestOnCancel}
          >Cancel</Button>,
          <Button
            key="submit" type="primary" size="large"
            loading={modalDialogSubmitInProgress}
            onClick={acceptRequestOnOK}
          >
            Accept
          </Button>,
        ]}
      >

</div>

Where the RequestInfo itself have:
<Row gutter={16} justify="center">
    <Col className={styles.parappa} span={24}>

    {expirationTime  ?
      <ReactCountdownClock seconds={calcCountDown(expirationTime)}
      color="#ff6600"
      timeFormat="hms"
      alpha={0.9}
      size={200}
      weight={20}

      onComplete={myCallback} />
      : ''
    }

    </Col>
    </Row>

After the Modal becomes visible and then invisible again, the clock speed goes crazy. Refresh page again, the clock speed is back to normal

**Comment by macabeus** [comment]
I have this same problem, but I do not know exactly in what situations this occurs.

**Comment by pughpugh** [comment]
Hi @macabeus . So far I've been able to replicate one instance where sending new props could double up the tick events. I believe this was fixed in `2.1.0`.

I think @PierreAndreis problem is specific to multiple timers on the same page. It's also possible that the component is also no longer mounting and un-mounting cleanly. Both cases need to be investigated further. I'm hoping I can look at this over the xmas holidays.
